scripts/populate-update-server.ts

```diff
@@ -0,0 +1,288 @@
+ *   GITHUB_REPOSITORY: The GitHub owner/repository (from GitHub Actions).
+ *   GITHUB_SHA:        Commit hash (if creating a new release).
+ *   GITHUB_ACTOR:      User that triggered this, github.actor
+ *   RD_SETUP_EXE:      The installer (exe file) to upload.
+ *   RD_SETUP_MSI:      The installer (msi file) to upload.
+ *   RD_MACX86_ZIP:     The macOS (x86_64) zip archive to upload.
+ *   RD_MACARM_ZIP:     The macOS (aarch64) zip archive to upload.
+ *   RD_BUILD_INFO:     Build information ("latest.yml" from electron-builder)
+ *   RD_OUTPUT_DIR:     Checkout of `gh-pages`, to be updated.
+ */
+
+import crypto from 'crypto';
+import fs from 'fs';
+import path from 'path';
+
+import { Octokit } from 'octokit';
+import yaml from 'yaml';
+
+import { spawnFile } from '@pkg/utils/childProcess';
+import { defined } from '@pkg/utils/typeUtils';
+
+/** Read input from the environment; throws an error if unset. */
+function getInput(name: string) {
+  const result = process.env[name];
+
+  if (!result) {
+    throw new Error(`Could not read input; \$${ name } is not set correctly.`);
+  }
+
+  return result;
+}
+
+/** Given a input variable that expects a single file, return it. */
+async function getInputFile(name: string) {
+  const inputPath = getInput(name);
+  const stat = await fs.promises.stat(inputPath);
+
+  if (!stat.isDirectory()) {
+    return inputPath;
+  }
+
+  for (const dirent of await fs.promises.readdir(inputPath, { withFileTypes: true })) {
+    if (dirent.isFile()) {
+      return path.join(inputPath, dirent.name);
+    }
+  }
+
+  throw new Error(`Could not find input file for ${ name }`);
+}
+
+/**
+ * assetInfo describes information we need about one asset.
+ */
+type assetInfo = {
+  /** file is the (full) path to the asset file. */
+  file: string;
+  /** filename is the base name of the asset. */
+  filename: string;
+  /** length of the file */
+  length: number;
+  /** checksum is the checksum file contents of the file. */
+  checksum: string;
+  /** checksumName is the base name of the checksum. */
+  checksumName: string;
+};
+
+/**
+ * Given environment name, write checksum contents for the file.
+ * @param name Name of the environment variable that holds the file path.
+ * @returns File name and checksum data.
+ */
+async function getChecksum(name: string, filenameOverride?: string): Promise<assetInfo> {
+  const filename = await getInputFile(name);
+  const outputName = filenameOverride || path.basename(filename);
+  const stat = await fs.promises.stat(filename);
+  const input = fs.createReadStream(filename);
+  const hasher = crypto.createHash('sha512');
+  const promise = new Promise((resolve) => {
+    input.on('end', resolve);
+  });
+
+  input.pipe(hasher).setEncoding('hex');
+  await promise;
+  await new Promise<void>((resolve) => {
+    hasher.end(() => {
+      resolve();
+    });
+  });
+
+  return {
+    file:         filename,
+    filename:     outputName,
+    length:       stat.size,
```

Comment:
Does the `fs.promises.stat(filename)` always return something successfully? is there a need for additional checks?